File: mobile/app/src/main/java/com/example/fripouilles/model/ContratGarde.kt
package com.example.fripouilles.model

import kotlinx.serialization.Serializable

@Serializable
data class ContratGarde(
    val id: Int,
    val enfantId: Int,
    val parentId: Int,
    val assistantId: Int,
    val dateDebut: String,
    val dateFin: String? = null,
    val statut: StatutContrat,
    val tarifHoraireBrut: String,
    val nombreHeuresSemaine: String,
    val indemniteEntretien: String,
    val indemniteRepas: String,
    val indemniteKm: String? = null,
    val enfant: EnfantInfo? = null
)

@Serializable
data class EnfantInfo(
    val id: Int,
    val nom: String,
    val prenom: String,
    val dateNaissance: String? = null  // Optionnel pour les ateliers
)

File: web/app/types/models/contrat-garde.ts
import { StatutContrat, Role, StatutValidation } from '../enums'
import { AssistantProfilModel, AssistantProfilCreateInput } from './assistant-profil'
import { EnfantModel, EnfantCreateInput } from './enfant'
import { ParentProfilModel, ParentProfilCreateInput } from './parent-profil'
import { PaieModel, PaieCreateInput } from './paie'
import { SuiviGardeAssistantModel, SuiviGardeAssistantCreateInput } from './suivi-garde-assistant'

export interface ContratGardeModel {
  id: number
  enfantId: number
  parentId: number
  assistantId: number
  dateDebut: string
  dateFin: string | null
  statut: StatutContrat
  tarifHoraireBrut: number
  nombreHeuresSemaine: number
  indemniteEntretien: number
  indemniteRepas: number
  indemniteKm: number | null
  revocationDemandeePar: Role | null
  revocationStatut: StatutValidation | null
  revocationDateDemande: string | null
  revocationMotif: string | null
  revocationDateValidation: string | null
  revocationCommentaireParent: string | null
  createdAt: string
  updatedAt: string
  assistant?: AssistantProfilModel
  enfant?: EnfantModel
  parent?: ParentProfilModel
  paies?: PaieModel[]
  suivis?: SuiviGardeAssistantModel[]
}

export interface ContratGardeCreateInput {
  id?: number
  enfantId?: number
  parentId?: number
  assistantId?: number
  dateDebut: string
  dateFin?: string | null
  statut: StatutContrat
  tarifHoraireBrut: number
  nombreHeuresSemaine: number
  indemniteEntretien: number
  indemniteRepas: number
  indemniteKm?: number | null
  revocationDemandeePar?: Role | null
  revocationStatut?: StatutValidation | null
  revocationDateDemande?: string | null
  revocationMotif?: string | null
  revocationDateValidation?: string | null
  revocationCommentaireParent?: string | null
  createdAt?: string
  updatedAt?: string
  assistant?: AssistantProfilCreateInput
  enfant?: EnfantCreateInput
  parent?: ParentProfilCreateInput
  paies?: PaieCreateInput[]
  suivis?: SuiviGardeAssistantCreateInput[]
}

export type ContratGardeUpdateInput = Partial<ContratGardeCreateInput>
